src/history/test/HistoryTests.cpp

```diff
@@ -512,6 +632,80 @@ TEST_CASE("History prefix catchup", "[history][catchup]")
     REQUIRE(b->getLedgerManager().getLastClosedLedgerNum() == 2 * freq + 7);
 }
 
+TEST_CASE("Catchup post-shadow-removal works", "[history]")
+{
+    uint32_t newProto = Bucket::FIRST_PROTOCOL_SHADOWS_REMOVED;
+    uint32_t oldProto = newProto - 1;
+
+    auto configurator =
+        std::make_shared<RealGenesisTmpDirHistoryConfigurator>();
+    CatchupSimulation catchupSimulation{VirtualClock::VIRTUAL_TIME,
+                                        configurator};
+
+    catchupSimulation.generateRandomLedger(oldProto);
+
+    // Different counts: with proto 12, catchup should adapt and switch merge
+    // logic
+    std::vector<uint32_t> counts = {0, std::numeric_limits<uint32_t>::max(),
+                                    60};
+
+    SECTION("Upgrade at checkpoint start")
```

Comment:
Again here I _think_ all 3 of these are checking the same condition(s) just with different ledgers -- so could probably just be a loop + some commentary explaining what's special about those 3 numbers -- but if they differ in other ways I'm not seeing it and would like to know!